// Copyright (c) 2016-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the BSD-style license found in the
// LICENSE file in the root directory of this source tree. An additional grant
// of patent rights can be found in the PATENTS file in the same directory.
syntax = "proto3";

import "orc8r/protos/common.proto";
import "orc8r/protos/service303.proto";

package magma.orc8r;
option go_package = "magma/orc8r/cloud/go/protos";

message StateID {
    string type = 1;
    string deviceID = 2;
}

message GetStatesRequest {
    string networkID = 1;
    repeated StateID ids = 2;
}

message GetStatesResponse {
    repeated State states = 1;
}

message ReportStatesRequest {
    repeated State states = 1;
}

message DeleteStatesRequest {
    string networkID = 1;
    repeated StateID ids = 2;
}

service StateService {
    rpc GetStates (GetStatesRequest) returns (GetStatesResponse) {}
    rpc ReportStates(ReportStatesRequest) returns (Void) {}
    rpc DeleteStates(DeleteStatesRequest) returns (Void) {}
}
